pkg/client/cache/namespace.go

```diff
@@ -0,0 +1,45 @@
+package cache
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/registry/namespace"
+
+	oapi "github.com/openshift/origin/pkg/api"
+	quotaapi "github.com/openshift/origin/pkg/quota/api"
+)
+
+type IndexerToNamespaceLister struct {
```

Comment:
> Shouldn't this lister be upstreamed?

Can I get a TODO?  I think we need to move the entire `InformerFactory` interface.